sql/catalyst/pom.xml

```diff
@@ -114,6 +114,10 @@
       <version>2.7.3</version>
       <type>jar</type>
     </dependency>
+    <dependency>
+      <groupId>org.apache.arrow</groupId>
+      <artifactId>arrow-vector</artifactId>
```

Comment:
Since the `ArrowColumnVector` is moved to the catalyst module, I have to move the dependency as well. I think it's fine as sql/core depends on sql/catalyst.